src/library/scala/util/Either.scala

```diff
@@ -339,6 +339,24 @@ sealed abstract class Either[+A, +B] extends Product with Serializable {
     case _        => this.asInstanceOf[Either[A1, B1]]
   }
 
+
+  /** Returns the right value if this is right
+    * or this value if this if this is left
+    *
+    * @example {{{
+    * val  l: Either[String, Either[String, Int]] = Left("pancake")
+    * val rl: Either[String, Either[String, Int]] = Right(Left("flounder"))
+    * val rr: Either[String, Either[String, Int]] = Right(Right(7))
+    *
+    *  l.flatten //Either[String, Int]: Left("pancake")
+    * rl.flatten //Either[String, Int]: Left("flounder")
+    * rr.flatten //Either[String, Int]: Right(7)
+    * }}}
+    * 
+    * Equivalent to `flatMap(id => id)`
+    */
+  def flatten[A1 >: A, B1](implicit ev: B <:< Either[A1, B1]): Either[A1, B1] = flatMap(b => ev(b))
```

Comment:
`flatMap(ev)` instead of `flatMap(b => ev(b))` ?
(`<:<` extends `scala.Function1` https://github.com/scala/scala/blob/v2.13.0-M4/src/library/scala/Predef.scala#L532 )